setup.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python
+
+import os
+import pathlib
+
+import yaml
+from jinja2 import Environment, FileSystemLoader, select_autoescape
+from packaging.requirements import Requirement
+from setuptools import setup
+
+
+def get_requirements():
+    """Get package requirements from conda recipe meta.yaml file"""
+    env = Environment(
+        loader=FileSystemLoader(pathlib.Path(__file__).parent / "recipe"),
+        autoescape=select_autoescape(),
+    )
+    template = env.get_template("meta.yaml")
+    meta = yaml.safe_load(template.render(environ=os.environ))
+
+    requirements = {}
+    for req in meta["requirements"]["run"]:
+        requirement = Requirement(req)
+        requirements[requirement.name] = str(requirement.specifier)
+
+    # Handle packages that have different names on conda-forge and PyPI
+    requirements["blosc"] = requirements.pop("python-blosc")
+    requirements["msgpack"] = requirements.pop("msgpack-python")
+
+    # Exclude packages not available on PyPI
+    del requirements["openssl"]
```

Comment:
Rather than deleting `python-graphviz` and adding `ipycytoscape` to the main conda recipe, why not swap them out here? Or, just delete `python-graphviz` and add `ipycytoscape` to an `extras_require` to avoid bringing in a moderately heavy dependency?

Reply:
I'm less worried about removing `python-snappy` and `nb_conda_kernels`, neither seem necessary for either package.

Reply:
Fair point. I switched everything to `ipycytoscape` in order to avoid discrepancies between `conda install coiled-runtime` and `pip install coiled-runtime`. These sorts of differences are the kind of things I could see confusing to users and potentially annoying for us to need to keep track of. 

How does `ipycytoscape` compare to other packages we already have included in the runtime? E.g. `pyarrow`, `numba`

Reply:
I guess I'm less worried about the size of the package so much as the relatively deep dependency stack, since it brings in a huge amount of the jupyter stack. When I do a `--force-reinstall` I get:
```bash
Collecting ipycytoscape
  Using cached ipycytoscape-1.3.3-py2.py3-none-any.whl (3.6 MB)
Collecting ipywidgets>=7.6.0
  Using cached ipywidgets-7.7.1-py2.py3-none-any.whl (123 kB)
Collecting spectate>=1.0.0
  Using cached spectate-1.0.1-py2.py3-none-any.whl (11 kB)
Collecting ipython>=4.0.0
  Using cached ipython-8.4.0-py3-none-any.whl (750 kB)
Collecting ipython-genutils~=0.2.0
  Using cached ipython_genutils-0.2.0-py2.py3-none-any.whl (26 kB)
Collecting jupyterlab-widgets>=1.0.0
  Using cached jupyterlab_widgets-1.1.1-py3-none-any.whl (245 kB)
Collecting widgetsnbextension~=3.6.0
  Using cached widgetsnbextension-3.6.1-py2.py3-none-any.whl (1.6 MB)
Collecting ipykernel>=4.5.1
  Using cached ipykernel-6.15.1-py3-none-any.whl (132 kB)
Collecting traitlets>=4.3.1
  Using cached traitlets-5.3.0-py3-none-any.whl (106 kB)
Collecting debugpy>=1.0
  Using cached debugpy-1.6.2-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (1.8 MB)
Collecting packaging
  Using cached packaging-21.3-py3-none-any.whl (40 kB)
Collecting nest-asyncio
  Using cached nest_asyncio-1.5.5-py3-none-any.whl (5.2 kB)
Collecting psutil
  Using cached psutil-5.9.1-cp38-cp38-manylinux_2_12_x86_64.manylinux2010_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (284 kB)
Collecting tornado>=6.1
  Using cached tornado-6.2-cp37-abi3-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (423 kB)
Collecting matplotlib-inline>=0.1
  Using cached matplotlib_inline-0.1.3-py3-none-any.whl (8.2 kB)
Collecting jupyter-client>=6.1.12
  Using cached jupyter_client-7.3.4-py3-none-any.whl (132 kB)
Collecting pyzmq>=17
  Using cached pyzmq-23.2.0-cp38-cp38-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (1.1 MB)
Collecting pygments>=2.4.0
  Using cached Pygments-2.12.0-py3-none-any.whl (1.1 MB)
Collecting pickleshare
  Using cached pickleshare-0.7.5-py2.py3-none-any.whl (6.9 kB)
Collecting prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0
  Using cached prompt_toolkit-3.0.30-py3-none-any.whl (381 kB)
Collecting stack-data
  Using cached stack_data-0.3.0-py3-none-any.whl (23 kB)
Collecting decorator
  Using cached decorator-5.1.1-py3-none-any.whl (9.1 kB)
Collecting setuptools>=18.5
  Using cached setuptools-63.4.1-py3-none-any.whl (1.2 MB)
Collecting jedi>=0.16
  Using cached jedi-0.18.1-py2.py3-none-any.whl (1.6 MB)
Collecting backcall
  Using cached backcall-0.2.0-py2.py3-none-any.whl (11 kB)
Collecting pexpect>4.3
  Using cached pexpect-4.8.0-py2.py3-none-any.whl (59 kB)
Collecting notebook>=4.4.1
  Using cached notebook-6.4.12-py3-none-any.whl (9.9 MB)
Collecting parso<0.9.0,>=0.8.0
  Using cached parso-0.8.3-py2.py3-none-any.whl (100 kB)
Collecting jupyter-core>=4.9.2
  Using cached jupyter_core-4.11.1-py3-none-any.whl (88 kB)
Collecting python-dateutil>=2.8.2
  Using cached python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
Collecting entrypoints
  Using cached entrypoints-0.4-py3-none-any.whl (5.3 kB)
Collecting terminado>=0.8.3
  Using cached terminado-0.15.0-py3-none-any.whl (16 kB)
Collecting nbformat
  Using cached nbformat-5.4.0-py3-none-any.whl (73 kB)
Collecting nbconvert>=5
  Using cached nbconvert-6.5.0-py3-none-any.whl (561 kB)
Collecting Send2Trash>=1.8.0
  Using cached Send2Trash-1.8.0-py3-none-any.whl (18 kB)
Collecting prometheus-client
  Using cached prometheus_client-0.14.1-py3-none-any.whl (59 kB)
Collecting argon2-cffi
  Using cached argon2_cffi-21.3.0-py3-none-any.whl (14 kB)
Collecting jinja2
  Using cached Jinja2-3.1.2-py3-none-any.whl (133 kB)
Collecting ptyprocess>=0.5
  Using cached ptyprocess-0.7.0-py2.py3-none-any.whl (13 kB)
Collecting wcwidth
  Using cached wcwidth-0.2.5-py2.py3-none-any.whl (30 kB)
Collecting pyparsing!=3.0.5,>=2.0.2
  Using cached pyparsing-3.0.9-py3-none-any.whl (98 kB)
Collecting asttokens
  Using cached asttokens-2.0.5-py2.py3-none-any.whl (20 kB)
Collecting pure-eval
  Using cached pure_eval-0.2.2-py3-none-any.whl (11 kB)
Collecting executing
  Using cached executing-0.9.1-py2.py3-none-any.whl (16 kB)
Collecting defusedxml
  Using cached defusedxml-0.7.1-py2.py3-none-any.whl (25 kB)
Collecting bleach
  Using cached bleach-5.0.1-py3-none-any.whl (160 kB)
Collecting tinycss2
  Using cached tinycss2-1.1.1-py3-none-any.whl (21 kB)
Collecting beautifulsoup4
  Using cached beautifulsoup4-4.11.1-py3-none-any.whl (128 kB)
Collecting nbclient>=0.5.0
  Using cached nbclient-0.6.6-py3-none-any.whl (71 kB)
Collecting jupyterlab-pygments
  Using cached jupyterlab_pygments-0.2.2-py2.py3-none-any.whl (21 kB)
Collecting mistune<2,>=0.8.1
  Using cached mistune-0.8.4-py2.py3-none-any.whl (16 kB)
Collecting pandocfilters>=1.4.1
  Using cached pandocfilters-1.5.0-py2.py3-none-any.whl (8.7 kB)
Collecting MarkupSafe>=2.0
  Using cached MarkupSafe-2.1.1-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (25 kB)
Collecting fastjsonschema
  Using cached fastjsonschema-2.16.1-py3-none-any.whl (22 kB)
Collecting jsonschema>=2.6
  Using cached jsonschema-4.9.1-py3-none-any.whl (79 kB)
Collecting six>=1.5
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting argon2-cffi-bindings
  Using cached argon2_cffi_bindings-21.2.0-cp36-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (86 kB)
Collecting pyrsistent!=0.17.0,!=0.17.1,!=0.17.2,>=0.14.0
  Using cached pyrsistent-0.18.1-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (119 kB)
Collecting importlib-resources>=1.4.0
  Using cached importlib_resources-5.9.0-py3-none-any.whl (33 kB)
Collecting pkgutil-resolve-name>=1.3.10
  Using cached pkgutil_resolve_name-1.3.10-py3-none-any.whl (4.7 kB)
Collecting attrs>=17.4.0
  Using cached attrs-22.1.0-py2.py3-none-any.whl (58 kB)
Collecting cffi>=1.0.1
  Using cached cffi-1.15.1-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (442 kB)
Collecting soupsieve>1.2
  Using cached soupsieve-2.3.2.post1-py3-none-any.whl (37 kB)
Collecting webencodings
  Using cached webencodings-0.5.1-py2.py3-none-any.whl (11 kB)
Collecting pycparser
  Using cached pycparser-2.21-py2.py3-none-any.whl (118 kB)
Collecting zipp>=3.1.0
  Using cached zipp-3.8.1-py3-none-any.whl (5.6 kB)
```

Reply:
Ah, I see. That's also a fair point. Though I think most of these packages are already included (indirectly) in the runtime today. If I create a fresh environment with just `coiled-runtime` (i.e. `mamba create --name foo python=3.9 coiled-runtime`) and then attempt to install `ipycytoscape` into that environment I only see three other dependencies installed:

```
❯ mamba install ipycytoscape

Looking for: ['ipycytoscape']

conda-forge/osx-64                                          Using cache
conda-forge/noarch                                          Using cache

Pinned packages:
  - python 3.9.*


Transaction

  Prefix: /Users/james/mambaforge/envs/foo

  Updating specs:

   - ipycytoscape
   - ca-certificates
   - certifi
   - openssl


  Package         Version  Build         Channel                  Size
────────────────────────────────────────────────────────────────────────
  Install:
────────────────────────────────────────────────────────────────────────

  + ipycytoscape    1.3.3  pyh1d7be83_0  conda-forge/noarch     Cached
  + networkx        2.8.5  pyhd8ed1ab_0  conda-forge/noarch     Cached
  + spectate        1.0.1  pyhd8ed1ab_0  conda-forge/noarch     Cached
  + traittypes      0.2.1  pyh9f0ad1d_2  conda-forge/noarch     Cached

  Summary:

  Install: 4 packages

  Total download: 0 B

────────────────────────────────────────────────────────────────────────
```